### PR TITLE
TGP-2033: PRA remove todo comments

### PR DESCRIPTION
**Pull request by yohan-oc**

### CODE REVIEW

**Review by Ewan-James-Donovan — DISMISSED** [review]

**Review by liandra-giacomine — DISMISSED** [review]

File: app/connectors/AccreditationConnector.scala [hunk 1]
```diff
@@ -35,7 +35,6 @@ class AccreditationConnector @Inject() (config: Configuration, httpClient: HttpC
   private val clientIdAndAcceptHeaders                         =
     Seq("X-Client-ID" -> "tgp-frontend", "Accept" -> "application/vnd.hmrc.1.0+json")
 
-  //TODO call should go to data store instead of router directly
```
Comment (besscerule) [inline app/connectors/AccreditationConnector.scala:38]: are we not doing this TODO? I feel it should go to the data-store

File: app/controllers/CyaCategorisationController.scala [hunk 2]
```diff
@@ -96,8 +96,6 @@ class CyaCategorisationController @Inject() (
             ).flatten
           )
 
-          //TODO this definitely will not survive in the new world
```
Comment (besscerule) [inline app/controllers/CyaCategorisationController.scala:99]: why has this been removed - has it been fixed?

File: app/controllers/CyaUpdateRecordController.scala [hunk 3]
```diff
@@ -66,62 +66,59 @@ class CyaUpdateRecordController @Inject() (
...
+        case Left(errors)     =>
+          logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadCommodityCode(recordId))
       }
     }
 
-  def getCountryOfOriginAnswer(userAnswers: UserAnswers, recordId: String)(implicit
+  private def getCountryOfOriginAnswer(userAnswers: UserAnswers, recordId: String)(implicit
     request: Request[_]
   ): Future[String] =
     userAnswers.get(CountryOfOriginUpdatePage(recordId)) match {
```
Comment (besscerule) [inline app/controllers/CyaUpdateRecordController.scala:124]: do we need to check what happens if it doesn't have an answer?

File: app/controllers/HasCorrectGoodsController.scala [hunk 4]
```diff
@@ -81,7 +81,6 @@ class HasCorrectGoodsController @Inject() (
       }
     }
 
-  // TODO - this is still not functional, it is just to create the url. Implement this properly
```
Comment (besscerule) [inline app/controllers/HasCorrectGoodsController.scala:84]: have we implemented this properly?

File: app/controllers/HasSupplementaryUnitController.scala [hunk 5]
```diff
@@ -49,8 +49,6 @@ class HasSupplementaryUnitController @Inject() (
 
   def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
-      //TODO this definitely will not survive in the new world
```
Comment (besscerule) [inline app/controllers/HasSupplementaryUnitController.scala:52]: again have we done a fix for this?

File: app/models/UpdateGoodsRecord.scala [hunk 6]
```diff
@@ -56,55 +56,30 @@ object UpdateGoodsRecord {
 
   def buildGoodsDescription(
```
Comment (besscerule) [inline app/models/UpdateGoodsRecord.scala]: we should change these function names to validateX and not buildX as we are not actually building anything

File: app/services/CategorisationService.scala [hunk 7]
```diff
@@ -60,7 +60,7 @@ class CategorisationService @Inject() (
                                                 request.affinityGroup,
                                                 Some(recordId),
                                                 getGoodsRecordResponse.countryOfOrigin,
-                                                LocalDate.now() //TODO where does DateOfTrade come from??
```
Comment (besscerule) [inline app/services/CategorisationService.scala:63]: is this done?

File: app/services/CategorisationService.scala [hunk 8]
```diff
@@ -103,7 +103,7 @@ class CategorisationService @Inject() (
                                   request.affinityGroup,
                                   Some(recordId),
                                   getGoodsRecordResponse.countryOfOrigin,
-                                  LocalDate.now() //TODO where does DateOfTrade come from??
```
Comment (besscerule) [inline app/services/CategorisationService.scala:106]: is this done?

File: app/services/CategorisationService.scala [hunk 9]
```diff
@@ -121,7 +121,6 @@ class CategorisationService @Inject() (
     } yield updatedAnswers
   }
 
-  //TODO this will be refactored out in TGP-1600 but solves the immediate problem
```
Comment (besscerule) [inline app/services/CategorisationService.scala:124]: was it refactored in 1600?

File: app/services/CategorisationService.scala [hunk 10]
```diff
@@ -141,7 +140,7 @@ class CategorisationService @Inject() (
                                             request.affinityGroup,
                                             Some(recordId),
                                             getGoodsRecordResponse.countryOfOrigin,
-                                            LocalDate.now() //TODO where does DateOfTrade come from??
```
Comment (besscerule) [inline app/services/CategorisationService.scala:144]: is this done?

File: test/controllers/GoodsRecordsControllerSpec.scala [hunk 11]
```diff
@@ -401,39 +401,5 @@ class GoodsRecordsControllerSpec extends SpecBase with MockitoSugar {
         ).toString
       }
     }
-
-    "must redirect to Journey Recovery for a GET if no userAnswers existing data is found" in {
```
Comment (besscerule) [inline test/controllers/GoodsRecordsControllerSpec.scala:405]: why has this test been removed?

File: test/models/UpdateGoodsRecordSpec.scala [hunk 12]
```diff
@@ -54,18 +54,17 @@ class UpdateGoodsRecordSpec extends AnyFreeSpec with Matchers with TryValues wit
 
         val answers =
           UserAnswers(userAnswersId)
-            .set(GoodsDescriptionUpdatePage(testRecordId), "goods description")
+            .set(GoodsDescriptionUpdatePage(testRecordId), "good description")
```
Comment (besscerule) [inline test/models/UpdateGoodsRecordSpec.scala]: needs an 's' >_<

File: app/controllers/CyaUpdateRecordController.scala [hunk 13]
```diff
@@ -171,49 +177,67 @@ class CyaUpdateRecordController @Inject() (
...
     (identify andThen getData andThen requireData).async { implicit request =>
-      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, request.eori, recordId) match {
-        case Right(model) =>
-          auditService.auditFinishUpdateGoodsRecord(recordId, request.affinityGroup, model)
+      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, recordId) match {
+        case Right(commodity) =>
+          auditService.auditFinishUpdateGoodsRecord(
+            recordId,
+            request.affinityGroup,
+            UpdateGoodsRecord(request.eori, recordId, commodityCode = Some(commodity), category = Some(1))
```
Comment (darbyo) [inline app/controllers/CyaUpdateRecordController.scala:219]: Why do we always set category to 1. are there other options? Apologies I am missing context here :) same below for line 223

File: test/models/UpdateGoodsRecordSpec.scala (hunk 12 above)
Comment (yohan-oc) [reply test/models/UpdateGoodsRecordSpec.scala]: updated

File: app/connectors/AccreditationConnector.scala (hunk 1 above)
Comment (yohan-oc) [reply app/connectors/AccreditationConnector.scala:38]: It’s not needed right now.

File: test/controllers/GoodsRecordsControllerSpec.scala (hunk 11 above)
Comment (yohan-oc) [reply test/controllers/GoodsRecordsControllerSpec.scala:405]: It's irrelevant now

File: app/controllers/CyaUpdateRecordController.scala (hunk 13 above)
Comment (yohan-oc) [reply app/controllers/CyaUpdateRecordController.scala:219]: Yes. It should be `category = Some(1)` https://github.com/hmrc/trader-goods-profiles-frontend/pull/290/files#diff-0d95f1a32505dd8cdc88e6597445c3bcebc79eaf8d2832c9cb02b26466b5cd60R53

File: app/controllers/HasSupplementaryUnitController.scala (hunk 5 above)
Comment (yohan-oc) [reply app/controllers/HasSupplementaryUnitController.scala:52]: There is a separate refactoring ticket for this

File: app/services/CategorisationService.scala (hunk 9 above)
Comment (yohan-oc) [reply app/services/CategorisationService.scala:124]: There is a separate refactoring ticket for this

File: app/controllers/HasCorrectGoodsController.scala (hunk 4 above)
Comment (yohan-oc) [reply app/controllers/HasCorrectGoodsController.scala:84]: There is a separate refactoring ticket for this

File: app/services/CategorisationService.scala (hunk 7 above)
Comment (yohan-oc) [reply app/services/CategorisationService.scala:63]: It's irrelevant now

File: app/services/CategorisationService.scala (hunk 8 above)
Comment (yohan-oc) [reply app/services/CategorisationService.scala:106]: It's irrelevant now

File: app/controllers/CyaUpdateRecordController.scala (hunk 3 above)
Comment (yohan-oc) [reply app/controllers/CyaUpdateRecordController.scala:124]: Not needed

File: app/controllers/CyaCategorisationController.scala (hunk 2 above)
Comment (yohan-oc) [reply app/controllers/CyaCategorisationController.scala:99]: There is a separate refactoring ticket for this

File: app/models/UpdateGoodsRecord.scala (hunk 6 above)
Comment (yohan-oc) [reply app/models/UpdateGoodsRecord.scala]: Updated

**Review by besscerule — APPROVED** [review]

**Pull request merged**